### PR TITLE
Improve docs

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- improve main README.md file with description how to run the Helm-Broker
- remove unnecessary fields from the configs

**Related issue(s)**
Resolves: https://github.com/kyma-project/kyma/issues/5239

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,16 +15,173 @@ To set up the project, download these tools:
...
+
+After the Helm-Broker processing the addons configuration you can see ClusterServiceClasses:
+
+```bash
+kubectl get clusterserviceclass
+```
+
+Provision Redis instance:
+```bash
+kubectl apply -f contrib/redis-instance.yaml
```
Comment (PK85) [inline README.md]: Add to yaml also some input parameters.

File: README.md [hunk 2]
```diff
@@ -16,16 +15,173 @@ To set up the project, download these tools:
...
+```
+
+Now you can check the status of the Redis instance:
+```bash
+kubectl get serviceinstance
+```
```
Comment (PK85) [inline README.md]: Add yaml to create service binding. Explain what is expected content of the secret.

File: README.md [hunk 3]
```diff
@@ -16,16 +15,173 @@ To set up the project, download these tools:
 
 These Go and Dep versions are compliant with the `buildpack` used by Prow. For more details read [this](https://github.com/kyma-project/test-infra/blob/master/prow/images/buildpack-golang/README.md) document.
 
-## Development
+## Installation 
 
-Before each commit, use the `before-commit.sh` script, which tests your changes.
+To run Helm-Broker you need a Kubernetes cluster with Tiller and Service-Catalog installed. To setup Helm-Broker in Minikube with all necessary dependencies follow these steps:
```
Comment (PK85) [inline README.md]: add same link as helm, how to install minikube

File: README.md [hunk 4]
```diff
@@ -4,8 +4,7 @@
 
 The Helm Broker is a [Service Broker](https://kyma-project.io/docs/master/components/service-catalog/#service-brokers-overview) which exposes Helm charts as Service Classes in the [Service Catalog](https://kyma-project.io/docs/master/components/service-catalog/#overview-overview). To do so, the Helm Broker uses the concept of addons. An addon is an abstraction layer over a Helm chart which provides all information required to convert the chart into a Service Class.
 
-For more information, read the [Helm Broker documentation](https://kyma-project.io/docs/master/components/helm-broker/).
-
+For more information, read the [Helm Broker documentation](https://kyma-project.io/docs/master/components/helm-broker/). If you need Helm-Broker installed with all dependencies try [Kyma](https://kyma-project.io/).
 ## Prerequisites
```
Comment (PK85) [inline README.md]: are they up to date?

File: README.md [hunk 5]
```diff
@@ -16,16 +15,173 @@ To set up the project, download these tools:
...
+```bash
+kubectl apply -f contrib/redis-instance.yaml
+```
+
+Now you can check the status of the Redis instance:
+```bash
+kubectl get serviceinstance
+```
+
+### Run Helm-Broker local version
```
Comment (PK85) [inline README.md]: MOve this section to /docs folder. IN development section link to that that file.

File: README.md [hunk 6]
```diff
@@ -16,16 +15,173 @@ To set up the project, download these tools:
...
+| **APP_SERVICE_NAME** | Yes | | The name of the Kubernetes Service which is exposing the broker. |
+| **APP_CLUSTER_SERVICE_BROKER_NAME** | Yes | | The name of the ClusterServiceBroker resource which registers the Helm-Broker into the Service-Catalog. |
+| **APP_DEVELOP_MODE** | No | false | If `true` insecure HTTP based repositories are allowed. |
+| **APP_DOCUMENTATION_ENABLED** | No | false | Indicates if Helm-Broker should upload addons documentation to the [Headless CMS](https://kyma-project.io/docs/components/headless-cms/). |
+
+## Development
+
+Before each commit, use the `before-commit.sh` script, which tests your changes.
 
+Integration tests checks if all part of Helm-Broker works together. It starts `kube-apiserver` with `etcd`, setups the controller and the broker. Before you run it, please install [Kubebuilder](https://github.com/kubernetes-sigs/kubebuilder). You can run integration tests using the follwoing command:
```
Comment (PK85) [inline README.md]: Explain how to develop a change, build image push and update chart and install upgrade a chart.

File: README.md [hunk 7]
```diff
@@ -4,9 +4,113 @@
 
 The Helm Broker is a [Service Broker](https://kyma-project.io/docs/master/components/service-catalog/#service-brokers-overview) which exposes Helm charts as Service Classes in the [Service Catalog](https://kyma-project.io/docs/master/components/service-catalog/#overview-overview). To do so, the Helm Broker uses the concept of addons. An addon is an abstraction layer over a Helm chart which provides all information required to convert the chart into a Service Class.
 
-For more information, read the [Helm Broker documentation](https://kyma-project.io/docs/master/components/helm-broker/).
+For more information, read the [Helm Broker documentation](https://kyma-project.io/docs/master/components/helm-broker/). If you need Helm-Broker installed with all dependencies try [Kyma](https://kyma-project.io/).
```
Comment (klaudiagrz) [inline README.md]:
```suggestion
For more information, read the [Helm Broker documentation](https://kyma-project.io/docs/master/components/helm-broker/). If you want to use the Helm Broker with all dependencies, try out [Kyma](https://kyma-project.io/).
```

File: README.md [hunk 8]
```diff
@@ -4,9 +4,113 @@
...
 
-## Prerequisites
```
Comment (klaudiagrz) [inline README.md]: Please turn back this section. List the requirements to run the project and provide links.

File: README.md [hunk 9]
```diff
@@ -4,9 +4,113 @@
...
+## Installation 
+
+To run Helm-Broker you need a Kubernetes cluster with Tiller and Service-Catalog installed. To setup Helm-Broker in Minikube with all necessary dependencies follow these steps:
+If you don't have Minikube installed, follow the [document](https://kubernetes.io/docs/tasks/tools/install-minikube/).
```
Comment (klaudiagrz) [inline README.md]: Add this info to the Prerequisites section.

File: README.md [hunk 10]
```diff
@@ -4,9 +4,113 @@
...
+
+Run the Minikube:
+```bash
+minikube start
+``` 
+
+If you don't have helm installed already, download [the Helm CLI](https://github.com/kubernetes/helm#install).
```
Comment (klaudiagrz) [inline README.md]: Add this info to the Prerequisites section.

File: README.md [hunk 11]
```diff
@@ -4,9 +4,113 @@
 
 The Helm Broker is a [Service Broker](https://kyma-project.io/docs/master/components/service-catalog/#service-brokers-overview) which exposes Helm charts as Service Classes in the [Service Catalog](https://kyma-project.io/docs/master/components/service-catalog/#overview-overview). To do so, the Helm Broker uses the concept of addons. An addon is an abstraction layer over a Helm chart which provides all information required to convert the chart into a Service Class.
 
-For more information, read the [Helm Broker documentation](https://kyma-project.io/docs/master/components/helm-broker/).
+For more information, read the [Helm Broker documentation](https://kyma-project.io/docs/master/components/helm-broker/). If you need Helm-Broker installed with all dependencies try [Kyma](https://kyma-project.io/).
 
-## Prerequisites
+## Installation 
+
+To run Helm-Broker you need a Kubernetes cluster with Tiller and Service-Catalog installed. To setup Helm-Broker in Minikube with all necessary dependencies follow these steps:
```
Comment (klaudiagrz) [inline README.md]:
```suggestion
To run the Helm Broker, you need a Kubernetes cluster with Tiller and Service Catalog. Follow these steps to set up the Helm Broker on Minikube with all necessary dependencies:
```

File: README.md [hunk 12]
```diff
@@ -4,9 +4,113 @@
...
+If you don't have Minikube installed, follow the [document](https://kubernetes.io/docs/tasks/tools/install-minikube/).
+
+Run the Minikube:
```
Comment (klaudiagrz) [inline README.md]:
```suggestion
1. Run the Minikube:
```

File: README.md [hunk 13]
```diff
@@ -4,9 +4,113 @@
...
+```bash
+minikube start
+``` 
+
+If you don't have helm installed already, download [the Helm CLI](https://github.com/kubernetes/helm#install).
+Install Tiller into your cluster:
```
Comment (klaudiagrz) [inline README.md]:
```suggestion
2. Install Tiller into your cluster:
```

File: README.md [hunk 14]
```diff
@@ -4,9 +4,113 @@
...
+```bash
+helm init
+```
+
+Install Service-Catalog as a Helm chart:
```
Comment (klaudiagrz) [inline README.md]:
```suggestion
3. Install the Service Catalog as a Helm chart:
```

File: README.md [hunk 15]
```diff
@@ -4,9 +4,113 @@
...
+```bash
+helm repo add svc-cat https://svc-catalog-charts.storage.googleapis.com
+helm install svc-cat/catalog --name catalog --namespace catalog
+```
+
+Install Helm-Broker chart:
```
Comment (klaudiagrz) [inline README.md]:
```suggestion
4. Install the Helm Broker chart:
```

File: README.md [hunk 16]
```diff
@@ -4,9 +4,113 @@
...
+```bash
+helm install charts/helm-broker --name helm-broker --namespace helm-broker
+```
+## Usage
+
+This section explains how to use the Helm-Broker.
+
+### Provisioning services provided by the Helm-Broker
+
+If you have installed Helm-Broker with Service-Catalog, you can add an Addon repository and provision Redis instance.
```
Comment (klaudiagrz) [inline README.md]: Why do we use Redis example here? Can't we make it more generic?

Comment (piotrmiskiewicz) [reply README.md]: This is an example of usage. I wanted to show a real case.

File: README.md [hunk 17]
```diff
@@ -4,28 +4,170 @@
...
 
-## Development
+Before each commit, use the `before-commit.sh` script which tests your changes and builds binaries. If you want to run the Helm Broker locally, read [this](/docs/run-local.md) document.
+
+Integration tests check if all parts of the Helm-Broker work together. They start `kube-apiserver` with `etcd`, sets up the Controller and the Broker. Before you run it, install [Kubebuilder](https://github.com/kubernetes-sigs/kubebuilder). You can run integration tests using the following command:
+```bash
+make integration-test
+```
+
+If you want to build Docker images with your changes and push them to a registry, follow these steps:
```
Comment (klaudiagrz) [inline README.md]:
```suggestion
### Build Docker images

If you want to build Docker images with your changes and push them to a registry, follow these steps:
```

File: docs/run-local.md [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+APP_DOCUMENTATION_ENABLED=false \
+APP_TMP_DIR=/tmp APP_NAMESPACE=default \
+APP_SERVICE_NAME=helm-broker \
+APP_CONFIG_FILE_NAME=contrib/local-etcd-config.yaml \
+APP_CLUSTER_SERVICE_BROKER_NAME=helm-broker \
+APP_DEVELOP_MODE=true \
+go run cmd/controller/main.go -metrics-addr ":8081"
+```
+
+>**NOTE:** Not all features are available when you run the Helm Broker locally. All features which perform actions with Tiller do not work. Additionally, the Controller performs operations on ClusterServiceBroker/ServiceBroker resources, which needs the Service Catalog to work properly.
```
Comment (klaudiagrz) [inline docs/run-local.md]:
```suggestion
>**NOTE:** Not all features are available when you run the Helm Broker locally. All features which perform actions with Tiller do not work. Moreover, the Controller performs operations on ClusterServiceBroker/ServiceBroker resources, which needs the Service Catalog to work properly.
```

File: docs/run-local.md [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+
+>**NOTE:** You can run the Controller and the Broker configured with the in-memory storage (without etcd), but then the Broker cannot read data stored by the Controller:
```
Comment (klaudiagrz) [inline docs/run-local.md]:
```suggestion
You can run the Controller and the Broker configured with the in-memory storage, but then the Broker cannot read data stored by the Controller. To run the Broker and the Controller without etcd, run these commands:
```

**Pull request merged**